```bash
mypy.....................................................................Failed
- hook id: mypy
- duration: 79.93s
- exit code: 1

pandas/core/strings/object_array.py:67: error: "ObjectStringArrayMixin" has no attribute "dtype"  [attr-defined]
pandas/core/strings/object_array.py:273: error: "ObjectStringArrayMixin" has no attribute "dtype"  [attr-defined]
pandas/core/strings/object_array.py:467: error: "ObjectStringArrayMixin" has no attribute "dtype"  [attr-defined]
Found 3 errors in 1 file (checked 1441 source files)
```

--- a/pandas/core/strings/object_array.py
+++ b/pandas/core/strings/object_array.py
@@ -64,7 +64,7 @@ class ObjectStringArrayMixin(BaseStringArrayMethods):
         if dtype is None:
             dtype = np.dtype("object")
         if na_value is None:
-            na_value = self.dtype.na_value
+            na_value = self.dtype.na_value  # type: ignore[attr-defined]
 
         if not len(self):
             return np.array([], dtype=dtype)
@@ -270,7 +270,7 @@ class ObjectStringArrayMixin(BaseStringArrayMethods):
                 return x.get(i)
             elif len(x) > i >= -len(x):
                 return x[i]
-            return self.dtype.na_value
+            return self.dtype.na_value  # type: ignore[attr-defined]
 
         return self._str_map(f)
 
@@ -464,7 +464,7 @@ class ObjectStringArrayMixin(BaseStringArrayMethods):
 
     def _str_extract(self, pat: str, flags: int = 0, expand: bool = True):
         regex = re.compile(pat, flags=flags)
-        na_value = self.dtype.na_value
+        na_value = self.dtype.na_value  # type: ignore[attr-defined]
 
         if not expand:
 

